File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        seen={}

        for i in range(len(numbers)):

            compliment=target-numbers[i]

            if compliment in seen:

                return [seen[compliment]+1,i+1]
                
            seen[numbers[i]]=i
        

File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer,Integer> map = new HashMap<>();

        int[] out = new int[2];

        for(int i=0;i<nums.length;i++){

            if(!map.containsKey(target-nums[i])){

                map.put(nums[i],i);
                
            }
            else{
                out[0]=map.get(target-nums[i]);
                out[1]=i;
                
            }
        }
        return out;
    }
}
